lib/src/chat/base_bloc/base_bloc.dart

```diff
@@ -0,0 +1,40 @@
+import 'dart:async';
```

Comment:
Is this implementation different? https://gist.github.com/boeledi/3050f0ef1192a274c3a509fa1bb9856d#file-bloc_event_state-dart

This is from the guy I borrowed the bloc_provider. It will be nice to keep the same style of code

Reply:
Yeah, the implementation is the same. You mean use that code?

Reply:
Yes, use the same code